src/libifconfig_internal.c

```diff
@@ -39,6 +40,19 @@
 #include "libifconfig_internal.h"
 
 int
+ifconfig_getifaddrs(ifconfig_handle_t *h)
+{
+	int ret;
+
+	if (h->ifap == NULL) {
+		ret = getifaddrs(&h->ifap);
+		/*h->ifa_cursor = h->ifap;*/
+		return (ret);
+	} else
```

Comment:
I'm a bit concerned about caching this information. Cache invalidation is one of the two hard problems in computer science (cache invalidation, naming things and off-by-one errors).

Reply:
Fixed in 48e9dc4